packages/react/kibana_context/root/root_provider.tsx

```diff
@@ -6,17 +6,21 @@
  * Side Public License, v 1.
  */
 
+import { EuiErrorBoundary } from '@elastic/eui';
 import { I18nStart } from '@kbn/core-i18n-browser';
 import React, { FC } from 'react';
 
 import { KibanaEuiProvider, type KibanaEuiProviderProps } from './eui_provider';
 
 /** Props for the KibanaRootContextProvider */
 export interface KibanaRootContextProviderProps extends KibanaEuiProviderProps {
+  // TODO: clintandrewhall - this should *not* be optional, but must be for now to avoid breaking
+  // components using `KibanaThemeProvider` as a rendering context.
   /** The `I18nStart` API from `CoreStart`. */
-  i18n: I18nStart;
+  i18n?: I18nStart;
```

Comment:
@cee-chen I'm not convinced this is the right way to handle the deprecation.  We might consider your detect-warn-and-resolve approach.